### PR TITLE
Adds radial skin menu to joy mask, pensive, angry, flushed

### PR DESCRIPTION
**Pull request by dzahlus**
## About The Pull Request
what it says on the tin

## Why It's Good For The Game
Joy mask is peak comedy, but we want to express ourselves in other ways at times, this allows anyone to do so.

## Changelog
:cl:
add: Added radial menu to joy mask for alt reskins
imageadd: added pensive, angry and flushed sprites to joy mask
/:cl:

### CODE REVIEW

**Comment by R3dtail** [comment]
Fetish content

File: code/modules/clothing/masks/miscellaneous.dm [hunk 1]
```diff
@@ -78,10 +78,41 @@
 	speech_args[SPEECH_MESSAGE] = trim(message)
 
 /obj/item/clothing/mask/joy
-	name = "joy mask"
-	desc = "Express your happiness or hide your sorrows with this laughing face with crying tears of joy cutout."
+	name = "Emotional Mask"
+	desc = "Express your happiness or hide your sorrows with this modular cutout."
 	icon_state = "joy"
+	clothing_flags = ALLOWINTERNALS
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0,"energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100)
```
Comment (LetterN) [inline code/modules/clothing/masks/miscellaneous.dm]: armored now?

File: code/modules/clothing/masks/miscellaneous.dm [hunk 2]
```diff
@@ -78,10 +78,41 @@
...
+	var/choice = show_radial_menu(user, src, joymask_designs, custom_check = FALSE, radius = 36, require_near = TRUE)
+
+	if(src && choice && !user.incapacitated() && in_range(user,src))
+		icon_state = options[choice]
+		user.update_inv_wear_mask()
+		for(var/X in actions)
+			var/datum/action/A = X
+			A.UpdateButtonIcon()
+		to_chat(user, "<span class='notice'>Your Joy mask now has a [choice] Emotion!</span>")
+		return 1
```
Comment (LetterN) [inline code/modules/clothing/masks/miscellaneous.dm:114]:
```suggestion
	if(QDELETED(src) || !choice || !user.incapacitated() || !in_range(user, src))
		return FALSE
	icon_state = options[choice]
	user.update_inv_wear_mask()
	for(var/X in actions)
		var/datum/action/A = X
		A.UpdateButtonIcon()
	to_chat(user, "<span class='notice'>Your Joy mask now has a [choice] Emotion!</span>")
	return TRUE
```

File: code/modules/clothing/masks/miscellaneous.dm [hunk 3]
```diff
@@ -78,10 +78,41 @@
...
+		"Angry" = image(icon = src.icon, icon_state = "angry"),
+		)
+
+/obj/item/clothing/mask/joy/ui_action_click(mob/user)
+	if(!istype(user) || user.incapacitated())
+		return
+
+	var/static/list/options = list("Joy" = "joy", "Flushed" = "flushed", "Pensive" = "pensive","Angry" ="angry")
+
+	var/choice = show_radial_menu(user, src, joymask_designs, custom_check = FALSE, radius = 36, require_near = TRUE)
```
Comment (LetterN) [inline code/modules/clothing/masks/miscellaneous.dm:105]:
```suggestion
	var/choice = show_radial_menu(user, src, joymask_designs, radius = 36, require_near = TRUE)
```

**Review by LetterN — CHANGES_REQUESTED** [review]

File: code/modules/clothing/masks/miscellaneous.dm (hunk 1 above)
Comment (dzahlus) [reply code/modules/clothing/masks/miscellaneous.dm]: theres probably a less shitcode way of making it not burn to ashes or melt

File: code/modules/clothing/masks/miscellaneous.dm (hunk 2 above)
Comment (dzahlus) [reply code/modules/clothing/masks/miscellaneous.dm:114]: this is copypasta from mime mask/paper mask not touching this

File: code/modules/clothing/masks/miscellaneous.dm (hunk 3 above)
Comment (dzahlus) [reply code/modules/clothing/masks/miscellaneous.dm:105]: see above

**Comment by LordSaladin** [comment]
SUSpensive

File: code/modules/clothing/masks/miscellaneous.dm [hunk 4]
```diff
@@ -78,10 +78,41 @@
 	speech_args[SPEECH_MESSAGE] = trim(message)
 
 /obj/item/clothing/mask/joy
-	name = "joy mask"
-	desc = "Express your happiness or hide your sorrows with this laughing face with crying tears of joy cutout."
+	name = "Emotional Mask"
+	desc = "Express your happiness or hide your sorrows with this modular cutout."
 	icon_state = "joy"
+	clothing_flags = ALLOWINTERNALS
+	resistance_flags = FIRE_PROOF //let us have this i beg you, it's a medium size breathmask :(
```
Comment (DeltaFire15) [inline code/modules/clothing/masks/miscellaneous.dm]: No need for fireproofness for a literal papermask

File: html/changelog.html [hunk 5]
```diff
@@ -50,6 +50,27 @@
...
+				<li class="bugfix">Fixes the material insertion animation for protolathes occurring on the circuit printer</li>
+				<li class="bugfix">Fixes the material insertion animation not actually playing when it was started</li>
+				<li class="bugfix">Fixes the material insertion animation for iron not being made</li>
+				<li class="bugfix">Fixes the material insertion animation sometimes not picking the right material for sheets that have multiple different materials in them</li>
+			</ul>
+			<h3 class="author">necromanceranne updated:</h3>
+			<ul class="changes bgimages16">
+				<li class="bugfix">Properly segments all the augment sprites so that they correctly display.</li>
+				<li class="bugfix">Properly labels the augment limb icon sprites.</li>
+				<li class="bugfix">Fixes going over the defined limit of roundstart prosthetics</li>
```
Comment (DeltaFire15) [inline html/changelog.html]: Remove all of this

**Review by DeltaFire15 — CHANGES_REQUESTED** [review]
Did you localtest it?

Also please address these two things.

File: code/modules/clothing/masks/miscellaneous.dm (hunk 4 above)
Comment (dzahlus) [reply code/modules/clothing/masks/miscellaneous.dm]: :(

**Comment by dzahlus** [comment]
> Did you localtest it?
> 
> Also please address these two things.

yes I always do

**Pull request merged**